src/ddscxx/include/org/eclipse/cyclonedds/topic/datatopic.hpp

```diff
@@ -316,7 +319,7 @@ bool serialize_into(void *buffer,
   S str;
   str.set_buffer(calc_offset(buffer, CDR_HEADER_SIZE), buf_sz-CDR_HEADER_SIZE);
   return (write_header<T,S>(buffer)
-        && write(str, sample, as_key)
+        && write(str, sample, as_key ? key_mode::unsorted : key_mode::not_key)
```

Comment:
I think that it would make sense to use `key_mode` as an argument here, too, rather than `as_key`: here you have a conditional expression, and at the call sites you either have the inverse conditional or a constant.

Reply:
I will improve on this.